userspace/libsinsp/metrics_collector.h

```diff
@@ -0,0 +1,314 @@
+	SINSP_STATS_V2_N_THREADS, ///< Total number of threads currently stored in the sinsp state thread table, unit: count.
+	SINSP_STATS_V2_N_FDS, ///< Total number of fds currently stored across all threadtables associated with each active thread in the sinsp state thread table, unit: count.
+	SINSP_STATS_V2_NONCACHED_FD_LOOKUPS, ///< fdtable state related counters, unit: count.
+	SINSP_STATS_V2_CACHED_FD_LOOKUPS, ///< fdtable state related counters, unit: count.
+	SINSP_STATS_V2_FAILED_FD_LOOKUPS, ///< fdtable state related counters, unit: count.
+	SINSP_STATS_V2_ADDED_FDS, ///< fdtable state related counters, unit: count.
+	SINSP_STATS_V2_REMOVED_FDS, ///< fdtable state related counters, unit: count.
+	SINSP_STATS_V2_STORED_EVTS, ///< evt parsing related counters, unit: count.
+	SINSP_STATS_V2_STORE_EVTS_DROPS, ///< evt parsing related counters, unit: count.
+	SINSP_STATS_V2_RETRIEVED_EVTS, ///< evt parsing related counters, unit: count.
+	SINSP_STATS_V2_RETRIEVE_EVTS_DROPS, ///< evt parsing related counters, unit: count.
+	SINSP_STATS_V2_NONCACHED_THREAD_LOOKUPS, ///< threadtable state related counters, unit: count.
+	SINSP_STATS_V2_CACHED_THREAD_LOOKUPS, ///< threadtable state related counters, unit: count.
+	SINSP_STATS_V2_FAILED_THREAD_LOOKUPS, ///< threadtable state related counters, unit: count.
+	SINSP_STATS_V2_ADDED_THREADS, ///< threadtable state related counters, unit: count.
+	SINSP_STATS_V2_REMOVED_THREADS, ///< threadtable state related counters, unit: count.
+	SINSP_STATS_V2_N_DROPS_FULL_THREADTABLE, ///< Number of drops due to full threadtable, unit: count.
+	SINSP_STATS_V2_N_MISSING_CONTAINER_IMAGES, ///<  Number of cached containers (cgroups) without container info such as image, hijacked sinsp_container_manager::remove_inactive_containers() -> every flush snapshot update, unit: count.
+	SINSP_STATS_V2_N_CONTAINERS, ///<  Number of containers (cgroups) currently cached by sinsp_container_manager, hijacked sinsp_container_manager::remove_inactive_containers() -> every flush snapshot update, unit: count.
+	SINSP_MAX_STATS_V2
+};
+
+#ifdef __linux__
+
+namespace libs::metrics
+{
+
+template <typename T>
+double convert_memory(metrics_v2_value_unit source_unit, metrics_v2_value_unit dest_unit, T val)
+{
+	double factor = 1;
+	switch(source_unit)
+	{
+	case METRIC_VALUE_UNIT_MEMORY_BYTES:
+		factor = 1;
+		break;
+	case METRIC_VALUE_UNIT_MEMORY_KIBIBYTES:
+		factor = 1024.;
+		break;
+	case METRIC_VALUE_UNIT_MEMORY_MEGABYTES:
+		factor = 1024. * 1024.;
+		break;
+	default:
+		return 0;
+	}
+
+	double bytes_val = val * factor;
+	switch(dest_unit)
+	{
+	case METRIC_VALUE_UNIT_MEMORY_BYTES:
+		return bytes_val;
+	case METRIC_VALUE_UNIT_MEMORY_KIBIBYTES:
+		return std::round((bytes_val / 1024.) * 10.) / 10.; // round to 1 decimal
+	case METRIC_VALUE_UNIT_MEMORY_MEGABYTES:
+		return std::round((bytes_val / 1024. / 1024.) * 10.) / 10.; // round to 1 decimal
+	default:
+		return 0;
+	}
+}
+
+class metrics_converter
```

Comment:
As an abstract class this needs a virtual destructor.
```c++
    virtual ~metrics_converter() = default;
```

Reply:
Great catch, included in the last commit. Plus rebased once more.